src/ConnectedMode.UnitTests/Suppressions/SuppressionIssueStoreUpdaterTests.cs

```diff
@@ -0,0 +1,295 @@
+            writer.Verify(x => x.AddIssues(new[] { issue }, true), Times.Once);
+
+            server.Invocations.Should().HaveCount(1);
+            writer.Invocations.Should().HaveCount(1);
+        }
+
+        [TestMethod]
+        public async Task UpdateAll_RunOnBackgroundThread()
+        {
+            var callSequence = new List<string>();
+
+            var queryInfo = new Mock<IServerQueryInfoProvider>();
+            var threadHandling = new Mock<IThreadHandling>();
+
+            queryInfo.Setup(x => x.GetProjectKeyAndBranchAsync(It.IsAny<CancellationToken>()))
+                .Callback<CancellationToken>(x => callSequence.Add("GetProjectKeyAndBranchAsync"));
+
+            threadHandling.Setup(x => x.SwitchToBackgroundThread())
+                .Returns(new NoOpAwaitable())
+                .Callback(() => callSequence.Add("SwitchToBackgroundThread"));
+
+            var testSubject = CreateTestSubject(queryInfo.Object,
+                threadHandling: threadHandling.Object);
+
+            await testSubject.UpdateAllServerSuppressionsAsync();
+
+            queryInfo.Invocations.Should().HaveCount(1);
+            threadHandling.Invocations.Should().HaveCount(1);
+
+            callSequence.Should().ContainInOrder("SwitchToBackgroundThread", "GetProjectKeyAndBranchAsync");
+        }
+
+        [TestMethod]
+        public void UpdateAll_CriticalExpression_NotHandled()
+        {
+            var queryInfo = new Mock<IServerQueryInfoProvider>();
+            queryInfo.Setup(x => x.GetProjectKeyAndBranchAsync(It.IsAny<CancellationToken>()))
+                .Throws(new StackOverflowException("thrown in a test"));
+
+            var logger = new TestLogger(logToConsole: true);
+
+            var testSubject = CreateTestSubject(queryInfo.Object, logger: logger);
+
+            Func<Task> operation = testSubject.UpdateAllServerSuppressionsAsync;
+            operation.Should().Throw<StackOverflowException>().And.Message.Should().Be("thrown in a test");
+
+            logger.AssertPartialOutputStringDoesNotExist("thrown in a test");
+        }
+
+        [TestMethod]
+        public void UpdateAll_NonCriticalExpression_IsSuppressed()
+        {
+            var queryInfo = new Mock<IServerQueryInfoProvider>();
+            queryInfo.Setup(x => x.GetProjectKeyAndBranchAsync(It.IsAny<CancellationToken>()))
+                .Throws(new InvalidOperationException("thrown in a test"));
+
+            var logger = new TestLogger(logToConsole: true);
+
+            var testSubject = CreateTestSubject(queryInfo.Object, logger: logger);
+
+            Func<Task> operation = testSubject.UpdateAllServerSuppressionsAsync;
+            operation.Should().NotThrow<InvalidOperationException>();
+
+            logger.AssertPartialOutputStringExists("thrown in a test");
+        }
+
+        [TestMethod]
+        public void UpdateAll_OperationCancelledException_CancellationMessageLogged()
```

Comment:
Tests that we get the simpler "operation cancelled" message if a call is cancelled by a second call to "fetch all".